packages/core/admin/admin/src/content-manager/components/EditViewDataManagerProvider/reducer.js

```diff
@@ -34,212 +40,177 @@ const reducer = (state, action) =>
-            set(draftState, [...path, 'disconnect'], newDisconnectArray);
-          }
+        /**
+         * If the field is a single relation field we don't want to append
+         * we just want to replace the value.
+         */
+        if (toOneRelation) {
+          set(draftState, path, [value]);
+        } else {
+          const modifiedDataRelations = get(state, path);
+          const newRelations = [...modifiedDataRelations, value];
+          set(draftState, path, newRelations);
         }
 
         break;
       }
       case 'DISCONNECT_RELATION': {
         const path = ['modifiedData', ...action.keys];
-        const { value } = action;
-        const connectedRelations = get(state, [...path, 'connect']);
-        const disconnectedRelations = get(state, [...path, 'disconnect']);
-
-        if (!disconnectedRelations) {
-          set(draftState, [...path, 'disconnect'], []);
-        }
-
-        const nextValue = get(draftState, [...path, 'disconnect']);
-        nextValue.push(value);
+        const { id } = action;
+        const modifiedDataRelation = get(state, [...path]);
 
-        if (connectedRelations?.length) {
-          const existsInConnect = connectedRelations.find(
-            (connectValue) => connectValue.id === value.id
-          );
+        /**
+         * TODO: before merge make this performant (e.g. 1000 relations === long time)
+         */
+        const newRelations = modifiedDataRelation.filter((rel) => rel.id !== id);
 
-          if (existsInConnect) {
-            const newConnectArray = pull([...connectedRelations], existsInConnect);
-            set(draftState, [...path, 'connect'], newConnectArray);
-          }
-        }
+        set(draftState, path, newRelations);
 
         break;
       }
+      /**
+       * This action will be called when you open your entry (first load)
+       * but also every time you press publish.
+       */
       case 'INIT_FORM': {
-        const { initialValues, relationalFields = [] } = action;
-
-        draftState.formErrors = {};
-
-        draftState.initialData = {
-          ...initialValues,
+        const {
+          initialValues,
+          relationalFields = [],
+          repeatableFields = [],
+          dynamicZones = [],
+        } = action;
 
-          /**
-           * The state we keep in the client for relations looks like:
-           *
-           * {
-           *   count: <int>
-           *   results: [<Relation>]
-           * }
-           *
-           * The content API only returns { count: <int> }, which is why
-           * we need to extend the existing state rather than overwriting it.
-           */
+        /**
+         * You can't mutate an actions value.
+         * and spreading an object only clones
+         * the first level, the deeply nested values
+         * are a reference.
+         */
+        const data = cloneDeep(initialValues);
```

Comment:
FYI: In the near future all browsers will have [`structuredClone()`](https://developer.mozilla.org/en-US/docs/Web/API/structuredClone).

Reply:
that's great, i think we need to get a "browserlist" to help us know when we can use these features 😄